suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -297,37 +275,39 @@ export const useAddCoinAccount = () => {
             return;
         }
 
-        const selectedType = accountType ?? NORMAL_ACCOUNT_TYPE;
+        const accounts = deviceAccounts.filter(
+            account => account.symbol === networkSymbol && account.accountType === accountType,
+        );
 
-        const canAddAccount = checkCanAddAccount({
-            networkSymbol,
-            accountType: selectedType,
-        });
+        const firstHiddenEmptyAccount = accounts.find(account => account.empty && !account.visible);
+
+        const canAddAccount = checkCanAddAccount(accounts);
 
         if (!canAddAccount) {
             return;
         }
 
-        const network = getNetworkToAdd({ networkSymbol, accountType: selectedType });
-
-        const accountsWithTypeLength = getAccountsForNetworSymbolkWithAccountType({
-            networkSymbol,
-            accountType: selectedType,
-        }).length;
+        const network = getNetworkToAdd({ networkSymbol, accountType });
 
         navigateToSuccessorScreen({
             flowType,
             networkSymbol,
-            accountType: selectedType,
-            accountIndex: accountsWithTypeLength,
+            accountType,
+            accountIndex: firstHiddenEmptyAccount?.index ?? accounts.length,
         });
 
-        const account = await dispatch(
-            addAndDiscoverNetworkAccountThunk({
-                network,
-                deviceState: device.state,
-            }),
-        ).unwrap();
+        if (firstHiddenEmptyAccount) {
+            dispatch(accountsActions.changeAccountVisibility(firstHiddenEmptyAccount));
+        }
+
+        const account =
+            firstHiddenEmptyAccount ||
+            (await dispatch(
+                addAndDiscoverNetworkAccountThunk({
+                    network,
+                    deviceState: device.state,
+                }),
+            ).unwrap());
```

Comment:
Temporary solution, I would expect removal of addAndDiscoverNetworkAccountThunk as part of https://github.com/trezor/trezor-suite/issues/11562

Reply:
Would it make sense to discover another hidden empty account here even in case this one is made visible?

Reply:
hmm, this probably does not make a sense unless the account is not empty.

But we can check regularly, that there exists at least one empty account of each coin-accountType pair and if not, run discovery until we find it.

Reply:
That is basically how it wokrks on desktop, but let's leave it please for another PR, this one is already bigger than expected 🙈

Reply:
for sure :)